### DISCUSSION TITLE
What are `combine` and `composite`?

### DISCUSSION THREAD

**Discussion by DOEHOONLEE** [ImageMagick/ImageMagick#4140] [Help] [answered]
I found these definitions on [ImageMagick(legacy)](https://legacy.imagemagick.org/script/command-line-options.php#compose).

```
// combine
The channels (previously set by -channel) of the combined image are taken from the grayscale values of each image in the sequence, in order. For the default -channel setting of RGB, this means the first image is assigned to the Red channel, the second to the Green channel, the third to the Blue.

This option can be thought of as the inverse to -separate, so long as the channel settings are the same. Thus, in the following example, the final image should be a copy of the original.

// compose
If a -compose method requires extra numerical arguments or flags these can be provided by setting the -set 'option:compose:args' appropriately for the compose method.
```
I do not understand what they mean though.

I saw this example on [StackOverflow](https://stackoverflow.com/questions/13232551/imagick-compose-with-mask) (there are sample images used too)

```
convert img.png \
\( dx.png dx.png dy.png -combine \) \
-define compose:args=312x26.6776 -compose displace -composite \
result2.png
```

So I get that by using `-combine` , it is taking the first image (dx.png) and assign it to the Red channel and the second image (dy.png) to the Green channel and there is no third one that should be assigned to the Blue.

What does it mean by assigning to Red/Green channel? and what is happening by assigning to those channels?

Also, what does `compose:args=312x26.6776` do? and what do the values mean here?

Thank you for reading a long question. Happy coding!

**Comment by fmw42** [comment, marked as answer]
It is taking dx.png and assigning it to the red channel and also to the green channel and taking dy and assigning it to the blue channel . Note that dx is there twice.  The compose:args tell imagemagick how much to do the displacement in the x and y directions by amplifying the values from the gray levels of the displacement map in the red, green and blue channels.

**Reply by DOEHOONLEE** [reply]
Alright alright. Much clearer now, but I guess I need to do some research on how the color channels work.

Thanks for helping me out!
